File: AB0-1-back/test_api.rb
# Test script para verificar endpoints da API
require 'net/http'
require 'json'
require 'uri'

puts "Testing API endpoints..."

# Test /api/v1/leads
begin
  uri = URI('http://localhost:3001/api/v1/leads')
  http = Net::HTTP.new(uri.host, uri.port)
  request = Net::HTTP::Get.new(uri)
  request['Accept'] = 'application/json'
  
  response = http.request(request)
  puts "✅ /api/v1/leads: Status #{response.code}"
  puts "Response: #{response.body[0..200]}..." if response.body
rescue => e
  puts "❌ /api/v1/leads: Error - #{e.message}"
end

# Test /api/v1/companies/5/analytics/traffic  
begin
  uri = URI('http://localhost:3001/api/v1/companies/5/analytics/traffic')
  http = Net::HTTP.new(uri.host, uri.port)
  request = Net::HTTP::Get.new(uri)
  request['Accept'] = 'application/json'
  
  response = http.request(request)
  puts "✅ /api/v1/companies/5/analytics/traffic: Status #{response.code}"
  puts "Response: #{response.body[0..200]}..." if response.body
rescue => e
  puts "❌ /api/v1/companies/5/analytics/traffic: Error - #{e.message}"
end

File: AB0-1-front/test-api.js
// Test script to verify companies API
fetch('http://localhost:3001/api/v1/companies')
  .then(response => response.json())
  .then(data => {
    console.log('Companies data:', data);
    console.log('Number of companies:', data.length);
    if (data.length > 0) {
      console.log('First company:', data[0]);
      console.log('First company has banner_url:', !!data[0].banner_url);
      console.log('First company has logo_url:', !!data[0].logo_url);
    }
  })
  .catch(error => {
    console.error('Error fetching companies:', error);
  });